### ISSUE TITLE
Keep component mounted

### ISSUE THREAD

**Issue by cedricpri** [inertiajs/inertia#281]
Hello! So, I recently moved my project to Laravel 8 and started looking at Inertia JS. I love it so far, thanks for the hard work!

However, there is one feature that I would like to be considered for this project: the possibility to keep a single component mounted, similar to vue router ```<keep-alive>``` tag. For example, I am displaying maps on my website and would like to avoid refreshing the map each time the page is changed, as keeping it alive would obviously decrease the loading time of each page and reduce the number of API calls needed in general.

I know you can achieve something similar with persistent layouts, the problem is that I would like the component to be reusable throughout the website and I would like to be able to display it in different places. As a workaround, I actually tried defining the following basic layout with my map, and I tried placing it to the correct location in another component with vue portal but it does not appear to be working, as the component is getting refreshed when the page changes.

    <template>
    <div>
        <portal to="myMap">
            <div slot-scope="props">
                <my-map :data="props.latLong"></my-map>
            </div>
        </portal>
        <slot></slot>
    </div>
    </template>

I am still quite new in web development, so I am not sure whether I am missing something and/or if implementing such a feature would be feasible? Thanks!

**Comment by reinink** [comment]
Hey @cedricpri, thanks for your interest in Inertia.js!

So, maybe I'm wrong, but this doesn't really sound like an Inertia issue, but more of a Vue "issue". I can't think of anyway to have persistent components in Vue that you can move around to different locations.

Are you suggesting that this is possible with Vue Router?

> the possibility to keep a single component mounted, similar to vue router tag

**Comment by cedricpri** [comment]
Exactly. I was actually mentioning the ```<keep-alive>``` tag shipped with vue router - it seems that Git removed the tag in the previous message since I didn't use markdown.

Anyway, the existence of the ```<router-view>``` in vue router made it easy to keep all the components mounted with this tag, which is exactly the kind of behavior I am trying to get with Inertia. Most of the content displayed being static, keeping everything alive when switching pages was quite interesting for me. It might be more a Vue related issue as you mention though.

**Comment by reinink** [comment]
Hmm, isn't `<keep-alive>` a Vue feature, not a Vue Router feature?

And, just so I'm clear in my understanding, you've actually done this with Vue Router? Can you show me an example?

**Comment by cedricpri** [comment]
I have just quickly build a JSFiddle showing what I mean: http://jsfiddle.net/rgdezqfn/46/. The thing is that, when using vue router, changing pages does not appear to destroy the components as the mounted method gets called only once. This is exactly what I need, as I don't want to destroy the components when browsing to a different page (by the way, I think ```<keep-alive>``` has nothing to do and this is just a default behavior of vue router, as removing the tag does not change anything).

My understanding was that the same was true for Inertia, that it was able to reach the backend, get the new page and render it without full page refresh (obviously), but also without destroying components when doing so. It doesn't seem to be the case though, as I observed when creating same two components:

    <template>
    <div>
        <inertia-link href="/foo"> <button> Go to Foo </button></inertia-link>
        <div>I am Bar.</div>
    </div>
    </template>

    <script>
    export default {
    mounted() {
        console.log("I have been mounted");
    }
    };
    </script>
    
and:

    <template>
    <div>
        <inertia-link href="/bar"> <button> Go to Bar </button></inertia-link>
        <div>I am Foo.</div>
    </div>
    </template>

    <script>
    export default {
    mounted() {
        console.log("I have been mounted");
    }
    };
    </script>

Then the mounted method gets called every time I switch between the components. I might be missing something obvious as I am just getting started with inertia and vue, but I wonder if this is intented and if there is any way around this?

**Comment by leeartem** [comment]
@cedricpri hello. found any solutions?

**Comment by reinink** [comment]
Hey! Thanks so much for your interest in Inertia.js and for sharing this issue/suggestion.

In an attempt to get on top of the issues and pull requests on this project I am going through all the older issues and PRs and closing them, as there's a decent chance that they have since been resolved or are simply not relevant any longer. My hope is that with a "clean slate" me and the other project maintainers will be able to better keep on top of issues and PRs moving forward.

Of course there's a chance that this issue is still relevant, and if that's the case feel free to simply submit a new issue. The only thing I ask is that you please include a super minimal reproduction of the issue as a Git repo. This makes it much easier for us to reproduce things on our end and ultimately fix it.

Really not trying to be dismissive here, I just need to find a way to get this project back into a state that I am able to maintain it. Hope that makes sense! ❤️
